pkg/bundle/sync.go

```diff
@@ -577,16 +584,17 @@ func (b *bundle) needsUpdate(ctx context.Context, log logr.Logger, obj *metav1.P
 	}
 
 	{
-		key := ""
-		targetFieldNames := []string{"data"}
-		targetType := obj.TypeMeta.Kind
```

Comment:
We shouldn't rely on the `TypeMeta.Kind` value being there as pointed out in https://github.com/kubernetes-sigs/controller-runtime/pull/2633.